src/core/layout/qgslayoutexporter.cpp

```diff
@@ -43,26 +57,693 @@ void QgsLayoutExporter::renderPage( QPainter *painter, int page )
   renderRegion( painter, paperRect );
 }
 
-void QgsLayoutExporter::renderRegion( QPainter *painter, const QRectF &region )
+QImage QgsLayoutExporter::renderPageToImage( int page, QSize imageSize, double dpi ) const
+{
+  if ( !mLayout )
+    return QImage();
+
+  if ( mLayout->pageCollection()->pageCount() <= page || page < 0 )
+  {
+    return QImage();
+  }
+
+  QgsLayoutItemPage *pageItem = mLayout->pageCollection()->page( page );
+  if ( !pageItem )
+  {
+    return QImage();
+  }
+
+  QRectF paperRect = QRectF( pageItem->pos().x(), pageItem->pos().y(), pageItem->rect().width(), pageItem->rect().height() );
+  return renderRegionToImage( paperRect, imageSize, dpi );
+}
+
+///@cond PRIVATE
+class LayoutItemCacheSettingRestorer
+{
+  public:
+
+    LayoutItemCacheSettingRestorer( QgsLayout *layout )
+      : mLayout( layout )
+    {
+      mLayout->context().mIsPreviewRender = false;
```

Comment:
`mLayout` could be checked to ensure that it's not null.

Reply:
i'll just assert here... there's absolutely no valid use for this object with no layout.

Reply:
Fixed